src/main/kotlin/stringcalculator/domain/customparser/CustomNumbersStringParser.kt

```diff
@@ -0,0 +1,48 @@
+package stringcalculator.domain.customparser
+
+class CustomNumbersStringParser(stringExpression: String, separators: ParserSeparators) {
+    val parsedNumbers: List<Int>
+
+    init {
+        validateSeparatorsEmpty(separators)
+        validateExpressionIncludeOtherStrings(stringExpression, separators)
+
+        parsedNumbers = splitBySeparators(stringExpression, separators)
+        validateParsedNumberForIncludeNegativeNumber(parsedNumbers)
+    }
+
+    private fun splitBySeparators(
+        stringExpression: String,
+        separators: ParserSeparators
+    ) = stringExpression.split(Regex(separators.getSeparatorsString().toString())).map { it.toInt() }
+
+    private fun validateSeparatorsEmpty(separators: ParserSeparators) {
+        require(separators.getSize() > 0) { ERROR_MESSAGE_SEPARATOR_ZERO }
+    }
+
+    companion object {
+        private const val ERROR_MESSAGE_SEPARATOR_ZERO = "숫자를 추출하기 위한 구분자가 없습니다"
+
+        private fun getErrorMessageIncludeNegativeNumber(negativeNumber: Int): String {
+            return "음수($negativeNumber)은 입력할수 없습니다"
+        }
+
+        fun validateParsedNumberForIncludeNegativeNumber(parsedNumbers: List<Int>) {
```

Comment:
접근제어자를 private로 해도 문제가 없어보이네요!
```suggestion
        private fun validateParsedNumberForIncludeNegativeNumber(parsedNumbers: List<Int>) {
```